src/backend/distributed/commands/owned.c

```diff
@@ -88,3 +88,36 @@ PreprocessDropOwnedStmt(Node *node, const char *queryString,
 
 	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
 }
+
+
+List *
+PreprocessReassignOwnedStmt(Node *node, const char *queryString,
+							ProcessUtilityContext processUtilityContext)
+{
+	ReassignOwnedStmt *stmt = castNode(ReassignOwnedStmt, node);
+	List *allReassignRoles = stmt->roles;
+
+	List *distributedReassignRoles = FilterDistributedRoles(allReassignRoles);
```

Comment:
Should remove filtering logic as we've discussed two weeks ago.

Reply:
We decided to keep the filtering logic

Reply:
Added tests for distributed and non-distributed roles with escape characters